### PR TITLE
feat: add LIC 9 (#13)

### LINKED ISSUE #13: LIC #9

### PR DESCRIPTION
**Pull request by spelgubbe**
Closes #13

### CODE REVIEW

**Comment by spelgubbe** [comment]
I think it's a mistake that I have written:

```java
if (angle < (PI - epsilon) || angle > (PI + epsilon))
{
  return true;
}
```
instead of using `doubleCompare`...

File: src/main/java/com/group12/LaunchInterceptorConditionCollection.java [hunk 1]
```diff
@@ -291,6 +291,56 @@ public boolean LIC8(List<Point> points, int aPts, int bPts, double radius1) thro
...
+
+        if (cPts < 1 || dPts < 1) {
+            throw new IllegalArgumentException("Parameters cPts and dPts both have to be greater than or equal to 1");
+        }
+
+        if (cPts + dPts > points.size() - 3) {
+            throw new IllegalArgumentException("Parameters cPts and dPts are required to sum to at most NUMPOINTS - 3");
+        }
+
+        if (epsilon < 0.0 || epsilon >= PI) {
```
Comment (nichujie) [inline src/main/java/com/group12/LaunchInterceptorConditionCollection.java]: Better to use `doubleCompare` here, especially for `>=`

File: src/main/java/com/group12/LaunchInterceptorConditionCollection.java [hunk 2]
```diff
@@ -594,4 +644,23 @@ private double angleFromThreePoints(Point vertex, Point point1, Point point2) {
         double magb = Math.sqrt(Math.pow(bx, 2) + Math.pow(by, 2));
         return Math.acos((ax * bx + ay * by) / (maga * magb));
     }
+
+    private boolean threePointsFulfillLIC9(Point v, Point p1, Point p2, double epsilon){
```
Comment (nichujie) [inline src/main/java/com/group12/LaunchInterceptorConditionCollection.java]: It is a bit strange to have a special function for LIC9 only. You can generalize the logic and modify LIC2 to use this function as well. LIC2 is merged in #38

**Review by nichujie — CHANGES_REQUESTED** [review]
`doubleCompare` and public logic

File: src/main/java/com/group12/LaunchInterceptorConditionCollection.java (hunk 2 above)
Comment (spelgubbe) [reply src/main/java/com/group12/LaunchInterceptorConditionCollection.java]: yeah its from trying to reduce code complexity

**Review by nichujie — APPROVED** [review]

**Pull request merged**